internal/webhooks/vspheremachine.go

```diff
@@ -92,6 +92,17 @@ func (webhook *VSphereMachineWebhook) ValidateCreate(_ context.Context, raw runt
 			allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "guestSoftPowerOffTimeout"), spec.GuestSoftPowerOffTimeout, "should be greater than 0"))
 		}
 	}
+	for i, device := range spec.PciDevices {
+		if device.VGPUProfile == "" {
+			if device.DeviceID == nil || device.VendorID == nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have both deviceId and vendorId set"))
+			}
+		} else {
+			if device.DeviceID != nil || device.VendorID != nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have either deviceId + vendorId or vgpuProfile"))
+			}
+		}
```

Comment:
The first case could be that all is set to nothing, so both should have the same message.

Let's change it maybe the following way:

```suggestion
		if device.VGPUProfile != "" && device.DeviceID == nil && device.VendorID == nil {
			// Valid case for vGPU.
			continue
		}
		if device.VGPUProfile == "" && device.DeviceID != nil && device.VendorID != nil {
			// Valid case for PCI Passthrough.
			continue
		}
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "pciDevices", fmt.Sprintf("%d", i)), device, "should have either deviceId + vendorId or vgpuProfile set"))
```

Reply:
Also: let's make a shared `validatePCIDevices` function to use in both places (vspheremachine + vspheremachinetemplate).